src/btree/bt_cursor.c

```diff
@@ -404,7 +404,11 @@ __cursor_col_modify(WT_CURSOR_BTREE *cbt, WT_ITEM *value, u_int modify_type)
 static inline int
 __cursor_row_modify(WT_CURSOR_BTREE *cbt, WT_ITEM *value, u_int modify_type)
 {
+#ifdef HAVE_DIAGNOSTIC
```

Comment:
This approach is missing the column store restore? Instead of checking the inside the function, is it possible to check at the upper layers, so that it can work for both row and column stores?

Reply:
We can do the same thing inside the column store functions. Doing it out side in the upper layer is hard.